client/search.jsx

```diff
@@ -56,7 +59,8 @@ class Search extends React.Component {
         const cityId = data.suggestions[0].entities[0].destinationId;
         fetch(`/api/search/list/${cityId}/${this.state.ratingFilter}`).then(response => response.json()).then(data => {
           const hotelList = data.data.body.searchResults.results;
-          this.setState({ userInput: this.state.userInput, hotelList: hotelList, searchButtonClicked: 'hidden', isLoading: false, ratingFilter: this.state.ratingFilter });
+          location.hash = '#hotel-details';
+          this.setState({ userInput: this.state.userInput, json: data.data.body, hotelList: hotelList, searchButtonClicked: 'hidden', isLoading: false, ratingFilter: this.state.ratingFilter });
```

Comment:
Your components are starting to get complicated. I think you should reword your app to have three routes (and 3 page components.
`#search` renders the `<Search />`
`results?location=something&rating=something` renders the `<SearchResults />`
`details?hotelId` renders the `<HotelDetails />`

Each of these components should do its own data fetching (none stored in App).